### PR TITLE
[CI] Add weights verification jobs

### PR DESCRIPTION
**Pull request by s3krit**
Adds two jobs:

1.  A job that runs after the runtime benchmarking jobs (which are manually triggered prior to a release) that checks that all of the weights files for a given runtime have been updated recently. This is currently set to a 2 day range, but could probably be shrunk to just $TODAY and the day before (to account for benchmark runs that take place overnight). Exampels of the runs can be found here: https://github.com/paritytech/polkadot/actions/runs/4681684403 - note that the jobs failed because they correctly identified that the weights have not been updated since the last release
2. A job that uses https://github.com/ggwpez/substrate-weight-compare to compare the weights of a PR against the weights of the last release, commenting with some nicely-formatted tables on any newly added weights, and changes changes in pre-existing weights. See the bottom of this PR for examples of the output.

### CODE REVIEW

**Comment by s3krit** [comment]
bot rebase

**Comment by ggwpez** [comment]
Really cool @s3krit!  

I think you can use a `spoiler` to make it easier on the eyes. Also there is a `--no-color` flag to fix the formatting (hope it works :smile:).
```pre
<details><summary>Details</summary>
<p>
</p>
</details> 
```

File: .github/workflows/check-weights.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          LAST_RELEASE=$(curl -s https://api.github.com/repos/paritytech/polkadot/releases/latest | jq -r .tag_name)
+          echo "LAST_RELEASE=$LAST_RELEASE" >> $GITHUB_ENV
+      - name: Checkout current sources
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Check weight changes
+        shell: bash
+        run: |
+          cargo install --git https://github.com/ggwpez/substrate-weight-compare swc
```
Comment (alvicsam) [inline .github/workflows/check-weights.yml:43]: I think it's worth to pin version of the package

File: scripts/ci/github/verify_updated_weights.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
```
Comment (ggwpez) [inline scripts/ci/github/verify_updated_weights.sh:1]: One thing that comes to my mind is that we probably also want this in cumulus, so could be that the scripts should rather live in https://github.com/paritytech/pipeline-scripts/

Although on the other hand we should "soon" have our monorepo, in which case it needs to be shuffled around anyway.

**Review by ggwpez — APPROVED** [review]
Looks good, thanks!

It does actually confirm that the bench script is broken:  
```pre
pallet_balances.rs was not updated for the current date. Last update: 2022-11-15
pallet_balances_nis_counterpart_balances.rs was not updated for the current date. Last update: 2023-03-15
```
since it cannot deal with multiple instance pallets…

File: .github/workflows/check-weights.yml (hunk 1 above)
Comment (ggwpez) [reply .github/workflows/check-weights.yml]: You can just add `--rev 30d1cd6b33ecf2dac32d48cb277f9b4d2d69786a` (latest master commit) if you are not a docker fan.

**Review by alvicsam — APPROVED** [review]

**Comment by s3krit** [comment]
bot rebase

**Comment by s3krit** [comment]
bot merge

**Comment by ggwpez** [comment]
So how is the process for this @coderobe ?  
It looks like they should run every time when a release happens and the weights got updated. I just want to ensure that we dont ignore these errors (too often).  
Could you then please ping me when it fails on the next release?

**Pull request merged**